### PR TITLE
Add error message for AtRiskUser without KrogerStore

### LINKED ISSUE #65: No Kroger store nearby sad path
## Issue
- [ ] Style (non-breaking style code)
- [ ] Testing (add testing to code)
- [ ] Bug fix (change which fixes an issue)
- [ ] Refactor (improvement to the code)
- [ ] New feature (change which adds functionality)
- [x] Chore
## Issue Summary

decide the response with front end and refactor `atriskuser` resource accordingly

### PR DESCRIPTION
**Pull request by adumortier**
## Who worked on this

- [x] Alexis
- [ ] Maria
- [ ] Ed
- [ ] Cody

## Type of change

- [ ] Style (Non-breaking Style code)
- [ ] Testing (Non-breaking Testing code)
- [x] Bug fix or refactor (change which fixes an issue)
- [ ] New feature (change which adds functionality)

## Summary of the change.

- changed the `closest_store` method to return an error message if Kroger response is empty
- changed the `at_risk_user` `post` method to return an error if zipcode does not match any store

Link to issue:

This closes #65 

🐞 If it is a bug or issue fix, describe the issue that was fixed.

Note any questions or struggles as well.

## Checklist:

- [x] My code follows the Turing style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented my code if necessary

### CODE REVIEW

File: trackbasket_be/models/krogerservice.py [hunk 1]
```diff
@@ -22,68 +22,71 @@ def refresh_token(cls):
...
     token = Krogerservice.refresh_token()
     parameters = {'filter.zipCode.near': zipcode, 'filter.limit':1}
-    headers = {'Authorization': 'Bearer {}'.format(token)}  
+    headers = {'Authorization': 'Bearer {}'.format(token)}
     response = requests.get('https://api.kroger.com/v1/locations?', params=parameters, headers=headers)
+    # parsed_response = json.loads(response.text)
     parsed_response = response.json()
-#     return {'status code': response.status_code, 
-#             'response': response.text, 
-#             'content-type': response.headers.get('Content-Type'),
```
Comment (adumortier) [inline trackbasket_be/models/krogerservice.py:38]: this is for debugging purposes, commented for now

File: trackbasket_be/models/krogerservice.py [hunk 2]
```diff
@@ -22,68 +22,71 @@ def refresh_token(cls):
...
-              'city': parsed_response['data'][0]['address']['city'], 
-              'state': parsed_response['data'][0]['address']['state'], 
-              'zipcode': parsed_response['data'][0]['address']['zipCode'],
-              'latitude': parsed_response['data'][0]['geolocation']['latitude'],
-              'longitude': parsed_response['data'][0]['geolocation']['longitude']            
-            }
-  
+    if parsed_response['data'] == []:
+      return  { 'error': 'no store found for this zipcode' }
+    else:
```
Comment (adumortier) [inline trackbasket_be/models/krogerservice.py:50]: checks if the response contains data and returns an error if it does not

File: trackbasket_be/resources/at_risk_user.py [hunk 3]
```diff
@@ -18,11 +18,14 @@ def post(self, id):
...
       at_risk_user = AtRiskUser(at_risk_user_id=id, **data)
       store_info = Krogerservice.closest_store(data['zipcode'])
-      store = Store(**store_info, at_risk_user=at_risk_user)
-      db.session.add(at_risk_user)
-      db.session.add(store)
-      db.session.commit()
-      return at_risk_user.json(), 201
+      if store_info == {'error': 'no store found for this zipcode'}:
+        return {'data': { 'id': 'at_risk_user', 'attributes': {'error': "No Kroger store match AtRiskUser zipcode"} } }, 400
+      else:
```
Comment (adumortier) [inline trackbasket_be/resources/at_risk_user.py:23]: returns a custom error to the front end when the `zipcode` does not match a nearby KrogerStore

Comment (mronauli) [reply trackbasket_be/resources/at_risk_user.py:23]: looks like there are two different error messages

**Pull request merged**